codegen/src/Azure.Iot.Operations.ProtocolCompilerLib/T4/communication/rust/Error/t4/RustAggregateError.tt

```diff
@@ -0,0 +1,37 @@
+<#@ template language="C#" linePragmas="false" #>
+<#@ import namespace="Azure.Iot.Operations.ProtocolCompilerLib" #>
+/* Code generated by Azure.Iot.Operations.ProtocolCompilerLib v<#=System.Reflection.Assembly.GetExecutingAssembly().GetName().Version#>; DO NOT EDIT. */
+
+use std::error::Error;
+use std::fmt;
+
+use super::<#=this.schemaName.GetFileName(TargetLanguage.Rust)#>::<#=this.schemaName.GetTypeName(TargetLanguage.Rust)#>;
+
+impl fmt::Display for <#=this.schemaName.GetTypeName(TargetLanguage.Rust)#> {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        let mut err_count = 0;
+
+<# foreach (var innerNameSchema in this.innerNameSchemas) { #>
+
+        if let Some(<#=innerNameSchema.Item1.GetFieldName(TargetLanguage.Rust)#>) = &self.<#=innerNameSchema.Item1.GetFieldName(TargetLanguage.Rust)#> {
+            if err_count == 0 {
+                return write!(f, "{<#=innerNameSchema.Item1.GetFieldName(TargetLanguage.Rust)#>}");
+            }
+            err_count += 1;
+        }
+<# } #>
+
+        match err_count{
+            0 => write!(f, "No reportable property errors."),
+            1 => Ok(()),
```

Comment:
I'm a bit confused by this match - can you provide some details (and then include that explanation as comments in the generated code)?